app/src/app/api/route-builder.ts

```diff
@@ -49,14 +58,37 @@ export class RouteBuilder<Context> {
 
   static init() {
     return new RouteBuilder(
-      (req: NextRequest, ctx: BaseContext, headers: Headers) => {
+      async (
+        req: NextRequest,
+        ctx: BaseContext,
+        headers: Headers,
+      ): Promise<Result<[BaseContext, Headers], NextResponse>> => {
+        let body: undefined | RequestBody = undefined;
+        if (req.body) {
+          const text = await req.text();
+          if (text) {
+            if (req.headers.get("Content-Type") === "application/json") {
+              try {
+                body = JSON.parse(text);
+              } catch {
+                return err(badRequestResponse());
+              }
+            } else if (req.headers.get("Content-Type") === "text/plain") {
+              body = text;
+            }
+          }
+        }
```

Comment:
Check prevents a badRequestResponse when the content-type is application/json but the body is empty. This happens when req.json() is used because req.json() throws an error when the body is null, despite null being valid JSON.